str/runners/str_iterative_eh_runner.py

```diff
@@ -52,105 +59,116 @@ def main(
-    project(name: $dataset) {
-        sequencingGroups(id: {in_: $input_cpg_sids}) {
-        id
-        sample {
-            externalId
-        }
-        analyses(type: {eq: "cram"}, active: {eq: true}) {
-            output
-            timestampCompleted
-        }
-        }
-    }
-
-    }
-        """
     )
-    response = query(
-        cram_retrieval_query,
-        variables={'dataset': dataset, 'input_cpg_sids': input_cpg_sids},
-    )
-    crams_by_id = {}
-    for i in response['project']['sequencingGroups']:
-        for cram in i['analyses']:
-            # ignore archived CRAMs and ONT crams
-            if cram['output'] is None:
-                continue
-            if 'archive' in cram['output']:
-                continue
-            if 'ont' in cram['output']:
-                continue
-            crams_by_id[i['id']] = cram
 
-    if len(crams_by_id) != len(input_cpg_sids):
-        cpg_sids_without_crams = set(input_cpg_sids) - set(crams_by_id.keys())
-        logging.warning(
-            f'There were some samples without CRAMs: {cpg_sids_without_crams}'
-        )
+    # list of catalog files (multiple, if catalog is sharded)
+    catalog_files = list(to_path(variant_catalog).glob('*.json'))
+    catalog_files = [
+        str(gs_path) for gs_path in catalog_files
+    ]  # coverts into a string type
 
-    eh_regions = b.read_input(variant_catalog)
+    # track number of jobs running
     jobs = []
-    # Iterate over each sample to call Expansion Hunter
-    for cpg_sample_id, cram_obj in crams_by_id.items():
-        # Making sure Hail Batch would localize both CRAM and the correponding CRAI index
-        crams = b.read_input_group(
-            **{'cram': cram_obj['output'], 'cram.crai': cram_obj['output'] + '.crai'}
-        )
 
-        # Working with CRAM files requires the reference fasta
-        ref = b.read_input_group(
-            **dict(
-                base=ref_fasta,
-                fai=ref_fasta + '.fai',
-                dict=ref_fasta.replace('.fasta', '')
-                .replace('.fna', '')
-                .replace('.fa', '')
-                + '.dict',
-            )
-        )
+    # open sample-sex mapping file
+    with to_path(sample_id_file).open() as f:
+        # Iterate over each sample to call Expansion Hunter
+        for line in f:
+            cpg_id = line.split(',')[0]
+            sex = line.split(',')[1]
+            if cpg_id == 's':  # header line
+                continue
+            if sex == 'XY':
+                sex_param = 'male'
+            else:
+                sex_param = 'female'
+                # 'X' and 'ambiguous' karyotypic sex will be marked
+                # as female (ExpansionHunter defaults to female if
+                # no sex_parameter is provided)
+
+            # retrieve corresponding cram path
+            cram_retrieval_query = gql(
+                """
+                query MyQuery($dataset: String!,$cpg_id: [String!]) {
```

Comment:
```suggestion
                query MyQuery($dataset: String!,$cpg_id: [String!]!) {
```

I got a bit distracted with this one, and it might not be important. AFAIK the `!` indicates that the value is not allowed to be null;  `[String!]` can be null or a list, but all entries within it cannot be null, `[String!]!` must be a list (not null) full of real String values. In your case this is always a list of Strings, so `[String!]!` would be accurate but I'd defer to @illusional whether this is important

Reply:
https://graphql.org/learn/schema/#object-types-and-fields